package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -79,12 +88,13 @@ def __init__(self, filename, chemfiles_format="", **kwargs):
         """
         Parameters
         ----------
-        filename : str
-            trajectory filename
+        filename : chemfiles.Trajectory or str
```

Comment:
I am not sure I understand why passing a chemfiles.Trajectory should work. What would be the use case for this? Users creating chemfiles trajectories and then wanting to do analysis with MDA?

Reply:
Yeah, it's so you can open a chemfiles thing as you want then pass this as the trajectory MDA will use.  Passing a filename and specifying `format='CHEMFILES'` still works, but that's relying on the baked in way to open a trajectory